Sources/Apollo/ApolloStore.swift

```diff
@@ -50,14 +50,17 @@ public final class ApolloStore {
   /// Clears the instance of the cache. Note that a cache can be shared across multiple `ApolloClient` objects, so clearing that underlying cache will clear it for all clients.
   ///
   /// - Returns: A promise which fulfills when the Cache is cleared.
-  public func clearCache() -> Promise<Void> {
-    return Promise<Void> { fulfill, reject in
-      queue.async(flags: .barrier) {
-        self.cacheLock.withWriteLock {
-          self.cache.clear()
+  public func clearCache(callbackQueue: DispatchQueue = .main, completion: ((Result<Void, Error>) -> Void)? = nil) {
+    queue.async(flags: .barrier) {
+      self.cacheLock.withWriteLock {
+          self.cache.clearPromise()
         }.andThen {
-          fulfill(())
-        }
+          guard let completion = completion else {
```

Comment:
Maybe `apollo_performAsyncIfNeeded` could be extended with a guard for an optional callback?